src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncMethodInAsyncContext.cs

```diff
@@ -83,6 +84,20 @@ public override void Initialize(AnalysisContext context)
                     return;
                 }
 
+                INamedTypeSymbol? semaphoreSlimType = wellKnownTypeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingSemaphoreSlim);
+                INamedTypeSymbol? timeSpanType = wellKnownTypeProvider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTimeSpan);
+                INamedTypeSymbol intType = context.Compilation.GetSpecialType(SpecialType.System_Int32);
+                IFieldSymbol? timeSpanZero = timeSpanType?.GetMembers(nameof(TimeSpan.Zero))
+                    .OfType<IFieldSymbol>()
+                    .FirstOrDefault();
+                ISet<IMethodSymbol> semaphoreSlimWaitMethods = semaphoreSlimType
```

Comment:
* `ImmutableArray<T>` will be at least as fast as `ISet` for such a small collection
* Not all overloads are included here, so I suggested a clarifying name

```suggestion
                ImmutableArray<IMethodSymbol> semaphoreSlimWaitWithTimeoutMethods = semaphoreSlimType
```